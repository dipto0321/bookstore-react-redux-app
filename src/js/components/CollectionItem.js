import React from 'react';
import PropTypes from 'prop-types';
import Progress from './Progress';
import CurrentChapter from './CurrentChapter';
import ProgressUpdateBtn from './ProgressUpdateBtn';
import 'react-circular-progressbar/dist/styles.css';

const CollectionItem = (props) => {
  const { handleRemove, item } = props;
  const {
    book_id, title, authors, category, page_count, current_page,
  } = item;

  return (
    <div className="card">
      <div className="card-body">
        <div className="columns">
          <div className="book-info">
            <h4 className="genre">{category}</h4>
            <h3 className="title">{title}</h3>
            <p>
              by:&nbsp;
              {authors}
            </p>
            <div className="action">
              <button className="btn btn-link" type="button" onClick={() => handleRemove(book_id)}>
                Remove
              </button>
            </div>
          </div>
          <div className="progress">
            <Progress
              currentPage={Number(current_page)}
              pages={Number(page_count)}
            />
          </div>
          <div className="chapter-info">
            <CurrentChapter currentPage={String(current_page)} />
            <ProgressUpdateBtn id={item.book_id} />
          </div>
        </div>
      </div>
    </div>
  );
};

CollectionItem.propTypes = {
  item: PropTypes.instanceOf(Object).isRequired,
  handleRemove: PropTypes.func.isRequired,
};

export default CollectionItem;
