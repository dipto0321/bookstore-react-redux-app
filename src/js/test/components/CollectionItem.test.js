import React from 'react';
import { shallow } from 'enzyme';
import CollectionItem from '../../components/CollectionItem';

describe('Book component', () => {
  let wrapper;
  let book;
  let handleRemove;
  beforeEach(() => {
    book = {
      id: 1,
      title: 'Antiquities of the Jews',
      author: 'Josephus',
      chapters: '30',
      currentChapter: '',
      category: 'History',
    };
    handleRemove = jest.fn();
    wrapper = shallow(<CollectionItem book={book} handleRemove={handleRemove} />);
  });

  test('it should render the Book component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it should call handleRemove on button click', () => {
    wrapper.find('button').simulate('click');
    expect(handleRemove).toHaveBeenLastCalledWith(book.id);
  });
});
