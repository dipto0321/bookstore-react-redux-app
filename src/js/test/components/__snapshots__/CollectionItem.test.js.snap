// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book component it should render the Book component correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="columns"
    >
      <div
        className="book-info"
      >
        <h4
          className="genre"
        >
          History
        </h4>
        <h3
          className="title"
        >
          Antiquities of the Jews
        </h3>
        <p>
          by:Â 
        </p>
        <div
          className="action"
        >
          <button
            className="btn btn-link"
            onClick={[Function]}
            type="button"
          >
            Remove
          </button>
        </div>
      </div>
      <div
        className="progress"
      >
        <Progress
          currentPage={NaN}
          pages={NaN}
        />
      </div>
      <div
        className="chapter-info"
      >
        <CurrentChapter
          currentPage="undefined"
        />
        <ConnectFunction />
      </div>
    </div>
  </div>
</div>
`;
